tests/script/test_realm.py

```diff
@@ -149,3 +182,62 @@ async def test_realm_realmDestroyed_sandbox(websocket, context_id):
             "realm": ANY_STR,
         }
     } == response
+
+
+@pytest.mark.asyncio
+async def test_realm_realmDestroyed_worker(websocket, context_id, html):
+    worker_url = 'data:application/javascript,while(true){}'
+    url = html(f"<script>window.w = new Worker('{worker_url}');</script>")
+
+    await subscribe(websocket,
+                    ["script.realmDestroyed", "script.realmCreated"])
+
+    await send_JSON_command(
+        websocket, {
+            "method": "browsingContext.navigate",
+            "params": {
+                "context": context_id,
+                "url": url,
+                "wait": "complete",
+            }
+        })
+
+    # Wait for worker to be created
+    while True:
+        message = await read_JSON_message(websocket)
+        if message == {
+                "type": "event",
+                "method": "script.realmCreated",
+                "params": {
+                    "realm": ANY_STR,
+                    "origin": worker_url,
+                    "context": ANY_STR,
+                    "type": "dedicated-worker"
+                }
+        }:
+            break
+
+    # Then demolish it!
+    await send_JSON_command(
+        websocket, {
+            "method": "script.evaluate",
+            "params": {
+                "target": {
+                    "context": context_id
+                },
+                "expression": "window.w.terminate()",
+                "awaitPromise": True
+            }
+        })
+
+    # Wait for confirmation that worker was destroyed
+    while True:
```

Comment:
`wait_for_event`

Reply:
I don't think the `while True` here is required, as you terminate only one realm, so there should be only one event. Just `await wait_for_event` and assert the event is expected.